pkg/agent/agent_windows.go

```diff
@@ -76,6 +82,35 @@ func (i *Initializer) prepareHostNetwork() error {
 	return util.PrepareHNSNetwork(subnetCIDR, i.nodeConfig.NodeIPAddr, adapter)
 }
 
+// updateNodeMACAnnotation updates the Node's MAC address in the Annotations of the Node. The MAC address will be used
+// for direct routing via Openflow in noencap case.
+func (i *Initializer) updateNodeMACAnnotation() error {
+	if !i.networkConfig.TrafficEncapMode.SupportsNoEncap() {
+		return nil
+	}
+	patch, _ := json.Marshal(map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"annotations": map[string]string{
+				types.NodeMACAddressAnnotationKey: i.nodeConfig.UplinkNetConfig.MAC.String(),
+			},
+		},
+	})
+	var err error
+	for attempt := 0; attempt < 3; attempt++ {
+		_, err = i.client.CoreV1().Nodes().Patch(context.TODO(), i.nodeConfig.Name, apitypes.MergePatchType, patch, metav1.PatchOptions{})
```

Comment:
Is this a good place to use client-go's retry mechanism: https://github.com/kubernetes/client-go/blob/8c8fa70f7a2acc191d4787327639621f69992efd/examples/create-update-delete-deployment/main.go#L126?

Reply:
thanks for sharing it, updated.